normandy/recipes/models.py

```diff
@@ -174,18 +248,28 @@ class RecipeRevision(models.Model):
     name = models.CharField(max_length=255)
     action = models.ForeignKey('Action', related_name='recipe_revisions')
     arguments_json = models.TextField(default='{}', validators=[validate_json])
-    filter_expression = models.TextField(blank=False)
+    extra_filter_expression = models.TextField(blank=False)
+    channels = models.ManyToManyField(Channel)
+    countries = models.ManyToManyField(Country)
+    locales = models.ManyToManyField(Locale)
 
     @property
     def data(self):
         return {
             'name': self.name,
             'action': self.action,
             'arguments_json': self.arguments_json,
-            'filter_expression': self.filter_expression,
+            'extra_filter_expression': self.extra_filter_expression,
+            'channels': list(self.channels.all()),
+            'countries': list(self.countries.all()),
+            'locales': list(self.locales.all()),
         }
 
     @property
+    def filter_expression(self):
+        return self.extra_filter_expression
```

Comment:
Do you plan to make this concatenate everything together to add the locale/language/country filtering?